### PR TITLE
Add slipcover via grayskull

### PR DESCRIPTION
**Pull request by asford**
Add `slipcover` orginally via grayskull.

Required build patch to fix c++ compilation in conda-build environment.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by asford** [comment]
@conda-forge/help-python this is ready for review. 🙏

**Comment by asford** [comment]
@conda-forge/help-python-c 

I'm having some trouble here with the osx_64 build, which fails with:

`clang-14: error: the clang compiler does not support '-march=core2`

However, `-march=core2` is being set in CFLAGS in a standard `compiler("cxx")` build config.

Is this expected or unusual?

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=606149&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=1601

File: recipes/slipcover/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    sha256: 80433ab2ccc9d2cda69f3c5adca4d978c34f13efb551c36e32657598b8c207be
+  # missing file from 0.2.0 sdist required for build
+  # fixed in:
+  # https://github.com/plasma-umass/slipcover/pull/31
+  - path: pyptr.h
+  - patches:
+    - 0001-use-env-build-flags.patch
+
+build:
+  skip: true  # [py>=312 or py2k]
```
Comment (xhochy) [inline recipes/slipcover/meta.yaml:19]: Please omit this skip. It doesn't serve a (good) purpose on conda-forge and will interfere with the Python 3.12 migration once it is going to happen.
```suggestion
```

Comment (asford) [reply recipes/slipcover/meta.yaml:19]: Roger, thanks.

**Review by asford — COMMENTED** [review]
Fixed build pin, this ready for re-review.

File: recipes/slipcover/meta.yaml (hunk 1 above)
Comment (asford) [reply recipes/slipcover/meta.yaml]: I don't exactly know how to solve this. The package doesn't support py3.12 and has compiled components. Should I have version skips or version constraints in the file?

File: recipes/slipcover/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - setuptools
+  host:
+    - pip
+    - python >=3.8,<3.12
```
Comment (xhochy) [inline recipes/slipcover/meta.yaml:28]: As suggested by the linter. No need for skips, these are the current versions currently supported on conda-forge.
```suggestion
    - python
```

File: recipes/slipcover/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - 0001-use-env-build-flags.patch
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - setuptools
```
Comment (xhochy) [inline recipes/slipcover/meta.yaml:25]: This should be in host.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/slipcover/meta.yaml (hunk 3 above)
Comment (asford) [reply recipes/slipcover/meta.yaml]: Moved both to host

**Review by asford — COMMENTED** [review]
Revised, ready for rereview

File: recipes/slipcover/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  - patches:
+    - 0001-use-env-build-flags.patch
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - {{ compiler('cxx') }}
```
Comment (xhochy) [inline recipes/slipcover/meta.yaml:24]: Only `setuptools` should be in host, the compiler needs to stay in `build`.

Comment (asford) [reply recipes/slipcover/meta.yaml:24]: Moved compiler to build

**Pull request merged**